test/UnitTests/Microsoft.Testing.Platform.UnitTests/Helpers/SystemAsyncMonitorTests.cs

```diff
@@ -17,15 +17,19 @@ public async Task AsyncMonitor_ShouldCorrectlyLock()
         bool lockState = false;
         List<Task> tasks = [];
         var stopwatch = Stopwatch.StartNew();
-        for (int i = 0; i < 3; i++)
+        try
         {
-            tasks.Add(Task.Run(TestLock, TestContext.CancellationToken));
-        }
-
-        await Task.WhenAll([.. tasks]);
+            for (int i = 0; i < 3; i++)
+            {
+                tasks.Add(Task.Run(TestLock, TestContext.CancellationToken));
+            }
 
-        // Give more time to be above 3s
-        Thread.Sleep(500);
```

Comment:
This is an unnecessary delay. The test seems reliable without it.
If we found the test to be flaky in future, what we can do instead is adjust the assert below to use `Assert.IsGreaterThan(2500, stopwatch.ElapsedMilliseconds)`